examples/language-modeling/run_clm.py

```diff
@@ -341,9 +341,20 @@ def group_texts(examples):
             if (model_args.model_name_or_path is not None and os.path.isdir(model_args.model_name_or_path))
             else None
         )
-        trainer.train(model_path=model_path)
+        train_result = trainer.train(model_path=model_path)
         trainer.save_model()  # Saves the tokenizer too for easy upload
 
+        output_train_file = os.path.join(training_args.output_dir, "train_results.txt")
+        if trainer.is_world_process_zero():
+            with open(output_train_file, "w") as writer:
+                logger.info("***** Train results *****")
+                for key, value in sorted(train_result.metrics.items()):
+                    logger.info(f"  {key} = {value}")
+                    writer.write(f"{key} = {value}\n")
+
+            # Need to save the state, since Trainer.save_model saves only the tokenizer with the model
+            trainer.state.save_to_json(os.path.join(training_args.output_dir, "trainer_state.json"))
+
```

Comment:
oh, I didn't realize you were going to replicate my half-sync rather than the nice implementation - I only did a half-baked sync since I needed a quick fix, and there were several things that needed to be figured out - agreed upon - but that code wasn't great. that's why I said "starting to sync".

Why are we replicating this huge chunk of code when it can be a re-usable function. I don't think this goes against `transformers` code duplication - these are utils and aren't really contributing to understanding models.

I'd say a good model is:
https://github.com/huggingface/transformers/blob/37d6fb5d04e64ac50cabd18c03abeb82d0fc8a58/examples/seq2seq/finetune_trainer.py#L123-L136

But the other scripts have a slightly different logic, that's why it needs some tweaking.